File: algorithm/src/main/java/baekjoon/p10820/Main.java
package baekjoon.p10820;

import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        while (sc.hasNextLine()) {
            for (int solve : solve(sc.nextLine())) {
                System.out.print(solve + " ");
            }
            System.out.println();
        }
    }

    private static int[] solve(String input) {
        int[] ret = new int[4];

        for (char ch : input.toCharArray()) {
            if (ch >= 'a' && ch <= 'z') {
                ret[0]++;
            } else if (ch >= 'A' && ch <= 'Z') {
                ret[1]++;
            } else if (ch >= '0' && ch <= '9') {
                ret[2]++;
            } else {
                ret[3]++;
            }
        }

        return ret;
    }

}

File: algorithm/src/main/java/baekjoon/p10820/Main.kt
package baekjoon.p10820

fun main(args: Array<String>) {
    while (true) {
        val input = readLine() ?: return
        println(solve(input).joinToString(" "))
    }
}

private fun solve(input: String): IntArray {
    return intArrayOf(
            input.count { it.isLowerCase() },
            input.count { it.isUpperCase() },
            input.count { it.isDigit() },
            input.count { it.isWhitespace() }
    )
}